deploy/crds/infra.watch_serviceassurances_crd.yaml

```diff
@@ -42,6 +42,12 @@ spec:
             eventsEnabled:
               description: Whether the Service Assurance Operator should enable components related to events collection and storage.
               type: boolean
+            haEnabled:
+              description: Whether to deploy the services in HA mode.
+              type: boolean
+            esNodeCount:
```

Comment:
I would call this out at `elasticsearchNodeCount`

Reply:
Makes sense

Reply:
@csibbitt @leifmadsen Rather than letting the use specify a number of nodes for each component in HA mode, would it make more sense to actually toggle the HA mode with a pre-defined set of configs for HA and noHA? Or do we want to user to be able to turn these knobs?

Reply:
I'm kind of leaning towards the opinion that we should limit the number of knobs made available. I like the idea of `highAvailabilityEnabled` with a value of just `true` or `false` and then strictly defining what that means. It will limit our test surface as well.

We already have the ability for manifest overrides in the Operator, so I'm totally fine hiding a lot of these tunables and just giving `yes / no` setup options and defining strictly what was expect "HA" to mean.

Reply:
I think for now I tend to agree with leif that there should be exactly 1 knob; actually a switch. HA on or off. We will set the correct number and config of everything from there. As customer needs expand we might need to add more, but that seems like the best place to start.

Reply:
The only other consideration I could think of would be to allow it per deployment type, e.g.

* `metricsHighAvailabilityEnabled: {true|false}`
* `eventsHighAvailabilityEnabled: {true|false}`

Otherwise, our unstated goal has been to make the configuration as simple as possible, and to provide a biased configuration using information from testing. Providing a well defined set of simple configuration options with the ability to override that was the approach Chris originally recommended when discussing SAO, and I am a believer.

Reply:
I'd lean towards all or nothing in terms of HA i.e. no extra knobs per deployment type. If that is required later, we can add it later, but for now, metricsEnabled, eventsEnabled and highAvailabilityEnabled would seem sufficient.

Does the esNodeCount value need to remain in this file? Or should I use some hardcoded integers in the ansible files?
Indeed, what does this variable do here, and where is it set/manipulated?

Reply:
I don't think it needs to remain in the file. I'd just code it into the `defaults/main.yml` as a variable within the role, and then reference that from the default template at the top of the elasticsearch playbook.